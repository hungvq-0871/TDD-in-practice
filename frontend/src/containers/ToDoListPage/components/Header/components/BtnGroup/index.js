import React from 'react';
import PropTypes from 'prop-types'

import Button from 'components/Button'
import BtnGroupStyle from './index.style'

function BtnGroup({ onSearchClick, onExportClick, onShareClick, ...rest }) {
  return (
    <BtnGroupStyle {...rest}>
      <Button className="search-btn" onClick={onSearchClick}>Search</Button>
      <Button className="export-btn" onClick={onExportClick}>Export</Button>
      <Button className="share-btn" onClick={onShareClick}>Share</Button>
    </BtnGroupStyle>
  )
}

BtnGroup.propTypes = {
  onSearchClick: PropTypes.func.isRequired,
  onExportClick: PropTypes.func.isRequired,
  onShareClick: PropTypes.func.isRequired
}

export default BtnGroup;
