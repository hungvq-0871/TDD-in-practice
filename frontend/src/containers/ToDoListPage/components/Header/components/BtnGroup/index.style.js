import styled from 'styled-components';
import { ButtonGroup } from 'reactstrap';

const BtnGroupStyle = styled(ButtonGroup)`
  ${({ theme }) => (`
    margin-right: 15px;

    .btn {
      border-color: ${theme.grey1};
      color: ${theme.grey1};
      background-color: ${theme.white};
      outline: none;
      
      &:hover {
        border-color: ${theme.grey1};
        color: ${theme.grey1};
        background-color: ${theme.white};
      }
      
      &.btn-secondary {
        &:not(:disabled) {
          &:not(.disabled) {
            &:active {
              border-color: ${theme.grey1};
              color: ${theme.grey1};
              background-color: ${theme.white};
            }
            
            &:focus {
              box-shadow: none;
              border-color: ${theme.grey1};
              color: ${theme.grey1};
              background-color: ${theme.white};
            }
          }
        }
      }
    }
  `)}
`;

export default BtnGroupStyle;