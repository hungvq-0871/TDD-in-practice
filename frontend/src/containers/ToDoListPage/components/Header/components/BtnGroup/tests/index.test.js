import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import BtnGroup from '../index';

const renderBtnGroup = (onSearchClick, onExportClick, onShareClick) => {
  return render(<BtnGroup
    onSearchClick={onSearchClick}
    onExportClick={onExportClick}
    onShareClick={onShareClick}
  />)
}

describe('test how "BtnGroup" is rendered', () => {
  test("it should match it's snapshot", () => {
    const { container } = renderBtnGroup(() => { }, () => { }, () => { });

    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('test functions around "BtnGroup"', () => {
  test(`the "search btn" should execute it's onClick function on clicked`, () => {
    const handleOnClick = jest.fn(() => { });
    const { container } = renderBtnGroup(handleOnClick, () => { }, () => { });
    const searchBtnEle = container.querySelector('.search-btn');

    fireEvent.click(searchBtnEle);

    expect(handleOnClick.mock.calls.length).toBe(1);
  });

  test(`the "export btn" should execute it's onClick function on clicked`, () => {
    const handleOnClick = jest.fn(() => { });
    const { container } = renderBtnGroup(() => { }, handleOnClick, () => { });
    const exportBtnEle = container.querySelector('.export-btn');

    fireEvent.click(exportBtnEle);

    expect(handleOnClick.mock.calls.length).toBe(1);
  });

  test(`the "share btn" should execute it's onClick function on clicked`, () => {
    const handleOnClick = jest.fn(() => { });
    const { container } = renderBtnGroup(() => { }, () => { }, handleOnClick);
    const shareBtnEle = container.querySelector('.share-btn');

    fireEvent.click(shareBtnEle);

    expect(handleOnClick.mock.calls.length).toBe(1);
  });
});
