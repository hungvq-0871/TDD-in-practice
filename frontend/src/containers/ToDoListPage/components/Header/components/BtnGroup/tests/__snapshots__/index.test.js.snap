// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test how "BtnGroup" is rendered it should match it's snapshot 1`] = `
<div
  class="sc-bwzfXH SfLdw btn-group"
  role="group"
>
  <button
    class="sc-bdVaJa fiYuhs btn search-btn"
  >
    <i
      class="fas fa-search"
    />
  </button>
  <button
    class="sc-bdVaJa fiYuhs btn export-btn"
  >
    <i
      class="fas fa-file-upload"
    />
  </button>
  <button
    class="sc-bdVaJa fiYuhs btn share-btn"
  >
    <i
      class="fas fa-share-square"
    />
  </button>
</div>
`;
